adapters/cpmstar/cpmstar.go

```diff
@@ -125,11 +125,11 @@ func (a *Adapter) MakeBids(bidRequest *openrtb2.BidRequest, unused *adapters.Req
 	var errors []error
 
 	for _, seatbid := range bidResponse.SeatBid {
-		for _, bid := range seatbid.Bid {
+		for i := range seatbid.Bid {
 			foundMatchingBid := false
 			bidType := openrtb_ext.BidTypeBanner
 			for _, imp := range bidRequest.Imp {
-				if imp.ID == bid.ImpID {
+				if imp.ID == seatbid.Bid[i].ImpID {
```

Comment:
Understood that this is current flow. But I would highly recommend to use `openrtb2.Bid.MType` field here to get mediaTypeForImp.

Reply:
mtype is not a normal parameter we expect to receive from RTB  2.5 requests. According to the RTB 2.5 spec, we can only be sure to see banner, video.

Even if supported in RTB 2.6 or later, it does not seem like a good idea to make the code based on the presence of such parameters which are unlikely to be sent in earlier RTB version requests.

Reply:
We're not proposing any alterations to the RTB request. Instead, we're requesting a modification to the seatBid response to enable the identification of `adapters.TypedBid.BidType` based on the Mtype value set in the bidder response. 
I believe there won't be any backward compatibility concerns since we're not altering any existing aspects of the bidder response; you're solely populating the `openrtb2.Bid.MType` field in the response.

Reply:
I'm sorry but I am still not understanding.  

I agree that mtype can be available on the response, but wouldnt our ad-server endpoint which responds to the ultimate RTB request be the one responsible for populating this field, if it wants to? 

Not the bidder. This still seems outside the scope of the PBS bidder adapter.